File: CS/CustomRecurrenceFormDescendantSample/MyAppointmentEditForm.cs
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.UI;
using System.Windows.Forms;

namespace CustomRecurrenceFormDescendantSample
{
    public class MyAppointmentEditForm : AppointmentForm
    {
        RecurrenceType recurringType;

        public MyAppointmentEditForm(SchedulerControl control, Appointment apt,
            bool openRecurrenceForm, RecurrenceType type)
            : base(control, apt, openRecurrenceForm)
        {
            recurringType = type;
        }

        protected override Form CreateAppointmentRecurrenceForm(Appointment patternCopy,
            FirstDayOfWeek firstDayOfWeek)
        {
            return new MyAppointmentRecurrenceForm(patternCopy, firstDayOfWeek,
                this.Controller, this.recurringType);
        }
    }

}

File: VB/CustomRecurrenceFormDescendantSample/MyAppointmentEditForm.vb
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.UI
Imports System.Windows.Forms

Namespace CustomRecurrenceFormDescendantSample
	Public Class MyAppointmentEditForm
		Inherits AppointmentForm

		Private recurringType As RecurrenceType

		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment, ByVal openRecurrenceForm As Boolean, ByVal type As RecurrenceType)
			MyBase.New(control, apt, openRecurrenceForm)
			recurringType = type
		End Sub

		Protected Overrides Function CreateAppointmentRecurrenceForm(ByVal patternCopy As Appointment, ByVal firstDayOfWeek As FirstDayOfWeek) As Form
			Return New MyAppointmentRecurrenceForm(patternCopy, firstDayOfWeek, Me.Controller, Me.recurringType)
		End Function
	End Class

End Namespace
